local.js

```diff
@@ -103,7 +105,11 @@ function requestTokenDeduped(scope) {
 	return IN_FLIGHT_REQUESTS[scope];
 }
 
-module.exports = function getLocalJwt(scope) {
+module.exports = function getLocalJwt(scope, doNotKeepSessionAlive) {
+	if (!doNotKeepSessionAlive) {
+		doNotKeepSessionAlive = false;
+	}
+
```

Comment:
We had to invert the logic of the boolean variable responsible because passing `false` from upstream in Polymer is not really possible.

Reply:
And what I mean by that is, calling the variable `KeepSessionAlive` instead for example and defaulting to it being `true`. Polymer interprets a value assigned as being truthy, so upstream (in Sequences) when I tried a bunch of stuff to pass `false` to `polymer-frau-jwt` to in turn pass that here, it didn't work.

Here's some relevant Polymer documentation: https://polymer-library.polymer-project.org/1.0/docs/devguide/properties#configuring-boolean-properties

Reply:
polymer-frau-jwt does not need to store it as a property. It needn't be configurable there and can inline the value directly.

Reply:
I could inline it [here](https://github.com/Brightspace/polymer-frau-jwt/blob/7f664ad2a8df42890231ef7d8ddc7c5cd3e9c248/frau-jwt-local.js#L49) as you mention, but the issue arises when I pass the value from a user of polymer-frau-jwt, such as Sequences.


On [this line](https://github.com/BrightspaceHypermediaComponents/sequences/blob/42c5acf043a6ae987002c6261e97f8c7fb249a16/d2l-sequence-viewer/d2l-sequence-viewer.js#L344), I would change it from:

`<frau-jwt-local token="{{token}}" scope="*:*:* content:files:read content:topics:read content:topics:mark-read"></frau-jwt-local>` 
to 
`<frau-jwt-local token="{{token}}" scope="*:*:* content:files:read content:topics:read content:topics:mark-read" do-not-keep-session-alive="true"></frau-jwt-local>`

but there is no way to pass a false value down to frau-jwt-local, and then in turn to frau-jwt.

Do you know of an alternative approach perhaps?

Reply:
You don't need `frau-jwt-local` to accept it as a property. It should always always always pass `{ extendSession: false }` to request token. frau-jwt-local is a broken api and will cause the bug you're trying to fix here no matter where it's used if you don't send the parameter.

Reply:
Ahh I see. The intention was to be less invasive with these changes, i.e. I would like not to break functionality of other users of `frau-jwt-local` that expect it to behave as it does today in the buggy way. How do you feel about this reasoning? If you feel `frau-jwt-local` should always just pass `{ extendSession: false }` no matter what, then I am able to do that no problem (aside from the concern that there may be some unforeseen impacts). 🤔

Reply:
* Our original change was doing this to all users of frau-jwt everywhere, so this smaller set is better
* Can communicate to users of frau-jwt-local about the change
* Can express that frau-jwt-local is deprecated and should be moved away from in tools

But if this behaviour is a defect in one part of the product, it is also likely a defect in others, even if it is covering up other defects.

I would proceed with my suggestion, yes.

Reply:
Gotcha, that makes sense, I will make the suggested changes in an upcoming commit. Thanks!